content/apps/ProtonVPN IP Leakage on Linux and Workaround.md

```diff
@@ -0,0 +1,85 @@
+---
+title: "ProtonVPN IP Leakage on Linux and Workaround"
+date: 2022-10-08
+tags: ['Applications', 'Linux', 'Qubes OS', 'Privacy']
+author: Tommy
+---
+
+**Before We Start**... 
+
+I sent Proton an email regarding this issue in late August 2022 and was told they are working on fixing it, though it will take some time as it requires some architectural changes in how the killswitch works.
+
+## The Leak
+
+Ideally, when implementing a killswitch, a VPN client should drop all connections on non-VPN interfaces except when the connection is to the VPN provider's servers. This is necessary to prevent accidental leaks, at least by unprivileged applications. Unfortunately, the ProtonVPN client does not currently do this.
+
+Effectively, any application that binds to the public interface on your Linux system will expose your actual IP address when they connect to the internet regardless of whether you have the killswitch on, off, or "permanently" (persistently) enabled. This is problematic, especially for certain applications like Torrent clients, as they tend to use whatever interfaces they have access to (rather than just the default one) to connect to the internet
```

Comment:
```suggestion
Effectively, any application that binds to the connected physical interface (as opposed to the VPN's virtual interface) on your Linux system will expose your actual IP address, regardless of the killswitch state. This is problematic, especially for certain applications like Torrent clients, as they tend to use whatever interfaces they can access (rather than just the default one) to connect to the internet.
```